File: api/app/domain/services/prompts/plan.py
SYSTEM_PLAN_PROMPT = """
你是一个任务规划智能体（Task Planner Agent），你需要为任务创建或者更新计划：
1.分析用户消息并理解用户的需求
2.确定完成任务需要使用哪些工具
3.根据用户的消息确定工作语言
4.生成计划的目标和步骤
"""

# 创建Plan规划提示词模板
CREATE_PLAN_PROMPT = """
你现在正在根据用户的消息创建一个计划：
{message}

注意：
-**你必须使用用户消息中使用的语言执行问题**
-你的计划必须简洁明了，不要添加任何不必要的细节
-你的步骤必须是原子性且独立的，以便下一个执行者可以使用工具逐一执行他们
-

返回格式要求：
- 必须返回符合以下 Typescript 接口定义的 JSON格式
- 必须包含指定的所有必填字段
- 如果判断任务不可行，则"steps"返回空数组，"goal"返回空字符串

Typescript 接口定义：
```typescript
interface CreatePlanResponse {{
    /** 对用户消息的回复以及对任务的思考，尽可能详细，使用用户的语言 **/
    message: string;
    /** 根据用户消息确定的工作语言 **/
    language: string;
    /** 步骤数组，每个步骤包含id和描述 **/
    steps: Array<{{
        id: string;
        description: string;
    }}>;
    /** 根据上下文生成的计划目标 **/
    goal: string;
    /** 根据上下文生成的计划标题 **/
    title: string;
}}
```

JSON输出示例：
{{
    "message": "", 
    "lanuage": "", 
    "goal": "", 
    "title": "",
    steps: [
        {{
            "id": 1,
            "description": "步骤一描述"
        }}
    ] 
}}

输入：
- message：用户的消息
- attachments：用户的附件

输出：
- JSON格式的计划

用户消息：
{message}

附件:
{attachments}

"""

# 更新Plan规划提示词模板
UPDATE_PLAN_PROMPT = """


"""

File: api-node/src/domain/services/prompts/plan.ts
export const SYSTEM_PLAN_PROMPT = `
You are a task planning agent, and you need to create or update plans for tasks:
1. Analyze user messages and understand user requirements
2. Determine which tools are needed to complete the task
3. Determine the working language based on user messages
4. Generate plan objectives and steps
`.trim();

export const CREATE_PLAN_PROMPT = `
You are now creating a plan based on the user's message:
{message}

Notes:
- **You must use the language used in the user's message to execute the task**
- Your plan must be concise and clear, do not add any unnecessary details
- Your steps must be atomic and independent, so that the next executor can execute them one by one using tools
- You need to determine whether the task can be split into multiple steps; if yes, return multiple steps; otherwise, return a single step

Output format requirements:
- Must return JSON format that conforms to the following TypeScript interface definition
- Must include all specified required fields
- If the task is determined to be infeasible, return an empty array for "steps" and an empty string for "goal"

TypeScript interface definition:
\`\`\`typescript
interface CreatePlanResponse {
  message: string;
  language: string;
  steps: Array<{ id: string; description: string }>;
  goal: string;
  title: string;
}
\`\`\`

User message:
{message}

User attachments:
{attachments}
`.trim();

export const UPDATE_PLAN_PROMPT = `
You are updating the plan, and you need to update the plan based on the execution results of the step:
{step}

Notes:
- You can delete, add, or modify plan steps, but do not change the plan goal (goal)
- If the changes are minor, do not modify the description
- Only replan subsequent **unfinished** steps, do not change completed steps
- The output step IDs should start from the ID of the first unfinished step, and replan the steps thereafter
- If a step is completed or no longer necessary, please delete it
- Carefully read the step results to determine if it was successful; if not successful, change the subsequent steps
- Based on the step results, you need to update the plan steps accordingly

Output format requirements:
- Must return JSON format that conforms to the following TypeScript interface definition
- Must include all specified required fields

TypeScript interface definition:
\`\`\`typescript
interface UpdatePlanResponse {
  steps: Array<{ id: string; description: string }>;
}
\`\`\`

Step:
{step}

Plan:
{plan}
`.trim();
